components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/username/NotificationUsernameRecoveryManager.java

```diff
@@ -209,3 +101,138 @@
 //        }
 //    }
 //}
+
+    public boolean recoverUserByClaims(ArrayList<UserClaim> claims)
+            throws IdentityRecoveryException {
+
+        // Check whether username recovery enable
+        boolean isRecoveryUsernameEnable = Utils.getRecoveryConfigs().getRecovery().getUsername().isEnable();
+
+        if (!isRecoveryUsernameEnable) {
+
+            if (log.isDebugEnabled()) {
+                log.debug("Username Recovery is not enabled");
+            }
+            return false;
+            //TODO send exception
+//            throw Utils.handleClientException(
+//                    IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_USERNAME_RECOVERY_NOT_ENABLE, null);
+        }
+
+        if (claims == null || claims.size() < 1) {
```

Comment:
Shall we use [ArrayUtils.isEmpty()](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/ArrayUtils.html#isEmpty(java.lang.Object[])) instead